apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -53,21 +59,24 @@
 
   private static final Logger log = LoggerFactory.getLogger(ApiarySnsListener.class);
 
+  static final String PROTOCOL_VERSION = "1.0";
+
   private static final String TOPIC_ARN = System.getenv("SNS_ARN");
-  final static String PROTOCOL_VERSION = "1.0";
-  private final String protocolVersion = PROTOCOL_VERSION;
+
+  private final String tableParamFilter = System.getenv("TABLE_PARAM_FILTER");
+  private final Pattern tableParamFilterPattern;
 
   private final AmazonSNS snsClient;
 
   public ApiarySnsListener(Configuration config) {
-    super(config);
-    snsClient = AmazonSNSClientBuilder.defaultClient();
-    log.debug("ApiarySnsListener created ");
+    this(config, AmazonSNSClientBuilder.defaultClient());
   }
 
-  ApiarySnsListener(Configuration config, AmazonSNS snsClient) {
+  ApiarySnsListener(Configuration config, AmazonSNS snsClient) throws PatternSyntaxException {
```

Comment:
`PatternSyntaxException` is an unchecked RuntimeException so you don't need to put it here. If you want to then you should put it on the other constructor too to be consistent, or just remove it from here.

Reply:
Good point. Removed.